apps/api/e2e/tests/total-roi-calculation.test.ts

```diff
@@ -0,0 +1,352 @@
+import axios from "axios";
+import { sql } from "drizzle-orm";
+import { beforeEach, describe, expect, test } from "vitest";
+
+import { config } from "@/config/index.js";
+import { db } from "@/database/db.js";
+import {
+  CROSS_CHAIN_TRADING_TYPE,
+  StartCompetitionResponse,
+} from "@/e2e/utils/api-types.js";
+import { getBaseUrl } from "@/e2e/utils/server.js";
+import {
+  ADMIN_EMAIL,
+  ADMIN_PASSWORD,
+  ADMIN_USERNAME,
+  cleanupTestState,
+  createTestClient,
+  getStartingValue,
+  registerUserAndAgentAndGetClient,
+  wait,
+} from "@/e2e/utils/test-helpers.js";
+
+// Type definitions for database query results
+interface LeaderboardEntry {
+  pnl: string;
+  starting_value: string;
+}
+
+describe("Total ROI Calculation Tests", () => {
+  let adminApiKey: string;
+
+  beforeEach(async () => {
+    await cleanupTestState();
+
+    // Create admin account
+    const response = await axios.post(`${getBaseUrl()}/api/admin/setup`, {
+      username: ADMIN_USERNAME,
+      password: ADMIN_PASSWORD,
+      email: ADMIN_EMAIL,
+    });
+
+    adminApiKey = response.data.admin.apiKey;
+    expect(adminApiKey).toBeDefined();
+  });
+
+  test("should calculate totalRoi correctly for single competition", async () => {
+    // Setup admin client
+    const adminClient = createTestClient();
+    await adminClient.loginAsAdmin(adminApiKey);
+
+    // Register test agent
+    const { client: agentClient, agent } =
+      await registerUserAndAgentAndGetClient({
+        adminApiKey,
+        agentName: "ROI Test Agent",
+      });
+
+    // Create and start competition
+    const competitionName = `ROI Test Competition ${Date.now()}`;
+    const startResponse = (await adminClient.startCompetition({
+      name: competitionName,
+      description: "Test competition for ROI calculation",
+      agentIds: [agent.id],
+      tradingType: CROSS_CHAIN_TRADING_TYPE.DISALLOW_ALL,
+    })) as StartCompetitionResponse;
+    expect(startResponse.success).toBe(true);
+
+    const competitionId = startResponse.competition.id;
+    const startingValue = await getStartingValue(agent.id, competitionId);
+
+    const tradeAmount = 1000;
+    // Burn 1000 USDC (we have a known roi this way)
```

Comment:
These tests all rely on being able to burn and get a negative roi.  You'll see a lot of places where the roi is checked against `-1 * tradeAmount / startingValue`